### PR TITLE
VM Page

### PR DESCRIPTION
**Pull request by ghbjklhv1**
**Description**: Adds notice to using VM
**Why?**: Applications are inherently insecure, users should run extra-insecure apps in a VM.
`-` like what QubesOS does.

**What is the attached files**: I created a PR for this design. It doesn't need to be the final result but I wanted to get the ball rolling. Edits from maintainers are welcomed.

**Has this been discussed here**: Nope.

### CODE REVIEW

File: _includes/sections/vm.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+<div class="alert alert-warning" role="alert">
+  <strong> When opening up insecure applications use a VM!</strong>
+</div>
+
+<img src="/assets/img/layout/vm.jpg" width="367" height="369" class="img-fluid float-right" alt="virtual-machine-logo" style="margin-left:10px;">
+
+<ol>
+  <li><strong>Choosing a strong hyperviser.</strong>
+    <ul>
+      <li>Use one that is <a href="https://www.wikipedia.org/wiki/Free_software">free software</a> like QEMU</li>
```
[the remark this answers is not in the document]
Comment (ghbjklhv1) [reply _includes/sections/vm.html]: If I remember properly, Virtual box uses non-free code for GPU exceleration.

I've looked at others too, like GNOME (Linux) boxes and VMM (OpenBSD and UNIX-based OSes) but they only support a few platforms.

XEN also looked promising, but if I remember properly they don't work on any LIbreboot laptop and many Coreboot machines as they require non-free code for the CPU.

QEMU-kvm may require non-free code as well.

File: _includes/sections/vm.html [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      <li>Stay wary of KVM as it can be a security risk (accesses kernel).</li>
```
[the remark this answers is not in the document]
Comment (ghbjklhv1) [reply _includes/sections/vm.html]: KVM's in general widen your attack surface.

For more info just look up the micro vs monolithic kernel debate.
Gives you plenty of information and instances.

File: _includes/sections/vm.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <li>Choose one with managable settings like shared clipboard.</li>
```
[the remark this answers is not in the document]
Comment (ghbjklhv1) [reply _includes/sections/vm.html]: > Is shared clipboard a secure setting?

It is handy, but it generally should be turned off.

> manageable settings

I mean that you (the user) should be able to control it.

__________

Not sure what edit your recommending (if any) :)

File: _includes/sections/vm.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    </ul>
+  </li>
+
+  <li><strong>Choosing an OS.</strong>
+    <ul>
+      <li>View our <a href="https://www.privacytools.io/operating-systems/">OS section</a> for information on OSes that respect your privacy.</li>
```
[the remark this answers is not in the document]
Comment (ghbjklhv1) [reply _includes/sections/vm.html]: Tails is proprietary, not generally a good OS for stuff like this (uses Debian `non-free`).

- Whonix might be a good choice.

Qubes OS is iffy, it doesn't support most privacy focused hardware (Libreboot). It has non-free dependencies: https://github.com/QubesOS/qubes-issues/issues/5163

File: _includes/sections/vm.html [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    </ul>
+  </li>
+
+</ol>
+
+<h3>Related Information</h3>
+
+<ul>
+  <li><a href="https://security.stackexchange.com/questions/73338/can-virtual-machines-be-used-to-isolate-vulnerable-software">VMs Have Flaws</a> - Security Stack Echange question "Can virtual machines be used to isolate vulnerable software?".</li>
```
[the remark this answers is not in the document]
Comment (ghbjklhv1) [reply _includes/sections/vm.html]: If this is the general consensus, I'll remove it.

File: _includes/nav.html [hunk 6]
```diff
@@ -53,6 +53,8 @@
             <a class="dropdown-item" href="/software/networks/"><i class="fas fa-user-secret fa-fw"></i> Self-contained Networks</a>
             <a class="dropdown-item" href="/software/cloud/"><i class="fas fa-hdd fa-fw"></i> Self-Hosted Cloud Server</a>
             <a class="dropdown-item" href="/software/voip/"><i class="fas fa-phone fa-fw"></i> Video & Voice Messenger</a>
+            <a class="dropdown-item" href="/vm"><i class="fas fa-phone fa-fw"></i> Virtual Machines</a>
```
[the remark this answers is not in the document]
Comment (ghbjklhv1) [reply _includes/nav.html]: I thought it was Wikipedia which is generall licensed under CC.
However, I cannot refind it.

You may want to purge it before merging.
I'll try to add a new logo. How do I edit the color? All I get is this hot pink. XD

[the remark this answers is not in the document]
Comment (ghbjklhv1) [reply _includes/nav.html]: As for the nav, I'm thinking cubes:

https://fontawesome.com/icons/cubes?style=solid

As cubes has now become a method to explain isolation.
How do I add images to nav?

**Comment by Mikaela** [comment]
@blacklight447-ptio I am assigning this to you as I think this is more of your cup of tea.

File: _includes/nav.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (JonahAragon) [reply _includes/nav.html]: The `<i class="fas fa-phone fa-fw"></i>` is the icon, which you would want to change to `<i class="fas fa-cubes fa-fw"></i>`

**Comment by ghbjklhv1** [comment]
> I am unsure about this one, as it adds quite a bit of complicated information that can confuse non tech savvy users. maybe it would be more appropriate to make some sort of blog post about the topic on write.privacytools.io, and link to it for our advanced users who need a bit of extra security, and have the technical knowledge to run a VM?

@blacklight447-ptio VMs shouldn't be that hard, however I do understand your point.
Would it be better to recommend simplistic managers, like GNOME Boxes?

**Comment by JonahAragon** [comment]
Seems out of scope for this project as @blacklight447-ptio mentioned.

**Pull request closed without merging**